conans/client/rest/rest_client.py

```diff
@@ -85,11 +86,19 @@ def upload_package(self, pref, files_to_upload, deleted, retry, retry_wait):
         return self._get_api().upload_package(pref, files_to_upload, deleted, retry, retry_wait)
 
     def authenticate(self, user, password):
-        apiv1 = RestV1Methods(self.remote_url, self.token, self.custom_headers, self._output,
-                              self.requester, self.verify_ssl, self._put_headers)
-        # Use v1 for the auth because the "ping" could be also protected so we don't know if
-        # we can use v2
-        return apiv1.authenticate(user, password)
+        api_v1 = RestV1Methods(self.remote_url, self.token, self.custom_headers, self._output,
+                               self.requester, self.verify_ssl, self._put_headers)
+
+        if not self.refresh_token or not self.token:
+            if False and REFRESH_TOKEN in self._cached_capabilities[self.remote_url]:
```

Comment:
``if False``?

Reply:
Shit. I forgot to remove it.

Reply:
As you noticed, there is "no way" to test this because it is not implemented in the conan server. I could try some "unit-test" but at these level probably they will be pretty useless.

Reply:
I'm trying to add some kind of test.